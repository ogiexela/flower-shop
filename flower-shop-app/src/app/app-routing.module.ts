import { NgModule } from '@angular/core';
import {
  RouterModule,
  Routes,
} from '@angular/router';
import { ROOT_ROUTE } from './constants/routes/root.routes';

const routes: Routes = [
  {
    path: ROOT_ROUTE.FLOWERS,
    loadChildren: () => import('./pages/flowers/flowers.module').then(mod => mod.FlowersModule),
  },
  {
    path: '',
    redirectTo: ROOT_ROUTE.FLOWERS,
    pathMatch: 'full',
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
