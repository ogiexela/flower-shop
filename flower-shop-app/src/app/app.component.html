<header>
    <h1>FLOWERS</h1>
</header>
<nav>
    <a routerLink="/flowers/buy">buy</a>
    <a routerLink="/flowers/sell">sell</a>
</nav>
<main>
    <router-outlet></router-outlet>
</main>
