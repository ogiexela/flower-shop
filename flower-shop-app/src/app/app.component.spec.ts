import { HttpClientTestingModule } from '@angular/common/http/testing';
import {
  async,
  TestBed,
} from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { AppModule } from './app.module';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
      ],
      imports: [AppModule, HttpClientTestingModule, RouterTestingModule.withRoutes([])]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture.nativeElement).toMatchSnapshot('empty_state');
  });
});
