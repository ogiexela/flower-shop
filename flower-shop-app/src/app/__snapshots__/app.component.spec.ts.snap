// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app: empty_state 1`] = `
<div
  id="root0"
>
  <header>
    <h1>
      FLOWERS
    </h1>
  </header>
  <nav>
    <a
      routerlink="/flowers/buy"
    >
      buy
    </a>
    <a
      routerlink="/flowers/sell"
    >
      sell
    </a>
  </nav>
  <main>
    <router-outlet />
  </main>
</div>
`;
